### ISSUE TITLE
How can I merge 12 convolutional layers in the right way?

### ISSUE THREAD

**Issue by payman-kh** [fchollet/keras#4272]
I am trying to design a CNN which has 12 convolutional2D layers as the input layer (for 12 images). 
I then merge them using the 'Merge' function just as in the Keras documentation.  The output_shape is 27.

Using a batch_size of 50, I get the error:
Error: Incompatible shapes: [600,27] vs [50, 27]

It is like each of the 12 layers are producing an output separately (12*50 = 600)

what am I doing wrong? please help. This is my first keras project.

thanks a million!

**Comment by dandxy89** [comment]
What merge mode are you using?

```
mode: string or lambda/function. If string, must be one
of: 'sum', 'mul', 'concat', 'ave', 'cos', 'dot', 'max'. If lambda/function, it should take as input a list of tensors and return a single tensor.
```

Sounds like you are using 'concat'

**Comment by payman-kh** [comment]
yes I am using concat. How can I fix it?
after the convolutional layers I am using:

`    merged = Merge([conv1, conv2, conv3, conv4, conv5, conv6, conv7, conv8, conv9, conv10, conv11, conv12], mode='concat', concat_axis = 0)


    final_model = Sequential()
    final_model.add(merged)
    final_model.add(Dense(100, activation = 'relu'))
    #final_model.add(dropout())
    final_model.add(Dense(27, activation = 'softmax'))`

It is suppose to produce (50,27) but it is giving (600,27)

How can I fix it?

**Comment by dandxy89** [comment]
Take a look at these.

```
from keras.layers import merge, Convolution2D, MaxPooling2D, Input, Dense, Flatten
from keras.models import Model

# Basic Example 
input_img = Input(shape=(256, 256, 3))

# Define 3 Conv Layers 
tower_1 = Convolution2D(64, 1, 1, border_mode='same', activation='relu')(input_img)
tower_1 = Convolution2D(64, 3, 3, border_mode='same', activation='relu')(tower_1)

tower_2 = Convolution2D(64, 1, 1, border_mode='same', activation='relu')(input_img)
tower_2 = Convolution2D(64, 5, 5, border_mode='same', activation='relu')(tower_2)

tower_3 = MaxPooling2D((3, 3), strides=(1, 1), border_mode='same')(input_img)
tower_3 = Convolution2D(64, 1, 1, border_mode='same', activation='relu')(tower_3)

# Merge them using for example mode='sum'
merged = merge([tower_1, tower_2, tower_3], mode='sum', concat_axis=1)
merged = Flatten()(merged)

# Add some Dense Layers
out = Dense(1, activation='sigmoid')(merged)

# Create the Model
some_model = Model(input_img, out)
some_model.summary()
```

compared to

```
from keras.layers import merge, Convolution2D, MaxPooling2D, Input, Dense, Flatten
from keras.models import Model

# Basic Example 
input_img = Input(shape=(256, 256, 3))

# Define 3 Conv Layers 
tower_1 = Convolution2D(64, 1, 1, border_mode='same', activation='relu')(input_img)
tower_1 = Convolution2D(64, 3, 3, border_mode='same', activation='relu')(tower_1)

tower_2 = Convolution2D(64, 1, 1, border_mode='same', activation='relu')(input_img)
tower_2 = Convolution2D(64, 5, 5, border_mode='same', activation='relu')(tower_2)

tower_3 = MaxPooling2D((3, 3), strides=(1, 1), border_mode='same')(input_img)
tower_3 = Convolution2D(64, 1, 1, border_mode='same', activation='relu')(tower_3)

# Merge them using for example mode='sum'
merged = merge([tower_1, tower_2, tower_3], mode='concat', concat_axis=1)
merged = Flatten()(merged)

# Add some Dense Layers
out = Dense(1, activation='sigmoid')(merged)

# Create the Model
some_model = Model(input_img, out)
some_model.summary()
```

Note the mode parameter changing the shapes in the summary outputs.
